### PR TITLE
[ADLS] virtual network rule CRUD

### PR DESCRIPTION
**Pull request by lewu-msft**
## Description
Add virtual network rule CRUD to ADLS account
[desin review is here](https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/93)
## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceManager/DataLakeStore/Commands.DataLakeStore/Commands/RemoveAzureRmDataLakeStoreVirtualNetworkRule.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.DataLake.Store.Models;
+using Microsoft.PowerShell.Commands;
+using System.Collections.Generic;
+using System.IO;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.DataLakeStore
+{
+    [Cmdlet("Remove", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "DataLakeStoreVirtualNetworkRule", SupportsShouldProcess = true), OutputType(typeof(bool))]
```
Comment (cormacpayne) [inline src/ResourceManager/DataLakeStore/Commands.DataLakeStore/Commands/RemoveAzureRmDataLakeStoreVirtualNetworkRule.cs]: @lewu-msft same comment about adding the parameter set with the `-ResourceId` parameter.

In addition, you'll want to add a parameter set with an `-InputObject` parameter to allow users to pipe from the other cmdlets returning a virtual network rule object to this cmdlet.

For more information about this scenario, please see [the piping scenario here](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/piping-in-powershell.md#using-the-inputobject-parameter) and [the code sample of a `Remove-*` cmdlet here](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/patterns-for-standard-cmdlets.md#c-example-2).

File: src/ResourceManager/DataLakeStore/Commands.DataLakeStore/Commands/RemoveAzureRmDataLakeStoreVirtualNetworkRule.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                "Indicates a boolean response should be returned indicating the result of the delete operation."
+            )]
+        public SwitchParameter PassThru { get; set; }
+
+        [Parameter(Position = 4,
+            ValueFromPipelineByPropertyName = true, Mandatory = false,
+            HelpMessage = "Name of resource group under which want to retrieve the account.")]
+        [ResourceGroupCompleter()]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
```
Comment (cormacpayne) [inline src/ResourceManager/DataLakeStore/Commands.DataLakeStore/Commands/RemoveAzureRmDataLakeStoreVirtualNetworkRule.cs]: @lewu-msft same comment about if this parameter is necessary

File: src/ResourceManager/DataLakeStore/Commands.DataLakeStore/Commands/RemoveAzureRmDataLakeStoreVirtualNetworkRule.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    [Alias("Remove-AdlStoreVirtualNetworkRule")]
+    public class RemoveAzureRmDataLakeStoreVirtualNetworkRule : DataLakeStoreCmdletBase
+    {
+        [Parameter(ValueFromPipelineByPropertyName = true, Position = 0, Mandatory = true,
+            HelpMessage = "The Data Lake Store account to update the virtual network rule in")]
+        [ValidateNotNullOrEmpty]
+        [Alias("AccountName")]
+        public string Account { get; set; }
+
+        [Parameter(ValueFromPipelineByPropertyName = true, Position = 1, Mandatory = false,
```
Comment (cormacpayne) [inline src/ResourceManager/DataLakeStore/Commands.DataLakeStore/Commands/RemoveAzureRmDataLakeStoreVirtualNetworkRule.cs]: @lewu-msft this should be mandatory, correct? I don't believe the SDK supports providing a `null` virtual network rule name to the `Delete` call.

File: src/ResourceManager/DataLakeStore/Commands.DataLakeStore/Commands/SetAzureRmDataLakeStoreVirtualNetworkRule.cs [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+using Microsoft.Azure.Commands.DataLakeStore.Properties;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.DataLake.Store.Models;
+using Microsoft.PowerShell.Commands;
+using System.IO;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.DataLakeStore
+{
+    [Cmdlet("Set", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "DataLakeStoreVirtualNetworkRule", SupportsShouldProcess = true), OutputType(typeof(DataLakeStoreVirtualNetworkRule))]
```
Comment (cormacpayne) [inline src/ResourceManager/DataLakeStore/Commands.DataLakeStore/Commands/SetAzureRmDataLakeStoreVirtualNetworkRule.cs]: @lewu-msft same comments about adding the `-ResourceId` and `-InputObject` parameter sets.

In addition, [here's an code sample of a `Set-*` cmdlet](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/patterns-for-standard-cmdlets.md#c-example-3).

File: src/ResourceManager/DataLakeStore/Commands.DataLakeStore/Commands/SetAzureRmDataLakeStoreVirtualNetworkRule.cs [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        [ValidateNotNullOrEmpty]
+        [Alias("AccountName")]
+        public string Account { get; set; }
+
+        [Parameter(ValueFromPipelineByPropertyName = true, Position = 1, Mandatory = true,
+            HelpMessage = "The name of the virtual network rule.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(ValueFromPipelineByPropertyName = true, Position = 2, Mandatory = false,
```
Comment (cormacpayne) [inline src/ResourceManager/DataLakeStore/Commands.DataLakeStore/Commands/SetAzureRmDataLakeStoreVirtualNetworkRule.cs]: @lewu-msft same comment about removing the `Position` property from optional parameters (applies to other parameters in this cmdlet).

File: src/ResourceManager/DataLakeStore/Commands.DataLakeStore/help/Add-AzureRmDataLakeStoreFirewallRule.md [hunk 6]
```diff
@@ -37,7 +37,7 @@ This creates a new firewall rule called "MyRule" in account "ContosoADL" with an
 The Data Lake Store account to add the firewall rule to
 
 ```yaml
-Type: System.String
```
Comment (cormacpayne) [inline src/ResourceManager/DataLakeStore/Commands.DataLakeStore/help/Add-AzureRmDataLakeStoreFirewallRule.md]: @lewu-msft you'll need to regenerate the help using the [`Update-MarkdownHelpModule` cmdlet with the `-UseFullTypeName` parameter](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module) to ensure that the full type name is preserved in the markdown

File: src/ResourceManager/DataLakeStore/Commands.DataLakeStore/help/Add-AzureRmDataLakeStoreVirtualNetworkRule.md [hunk 7]
```diff
@@ -0,0 +1,155 @@
+---
+external help file: Microsoft.Azure.Commands.DataLakeStore.dll-Help.xml
+Module Name: AzureRM.DataLakeStore
+online version:
```
Comment (cormacpayne) [inline src/ResourceManager/DataLakeStore/Commands.DataLakeStore/help/Add-AzureRmDataLakeStoreVirtualNetworkRule.md]: @lewu-msft this value will need to be 

> online version: https://docs.microsoft.com/en-us/powershell/module/azurerm.datalakestore/add-azureatalakestorevirtualnetworkrule

File: src/ResourceManager/DataLakeStore/Commands.DataLakeStore/help/Add-AzureRmDataLakeStoreVirtualNetworkRule.md [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+schema: 2.0.0
+---
+
+# Add-AzureRmDataLakeStoreVirtualNetworkRule
+
+## SYNOPSIS
+{{Fill in the Synopsis}}
```
Comment (cormacpayne) [inline src/ResourceManager/DataLakeStore/Commands.DataLakeStore/help/Add-AzureRmDataLakeStoreVirtualNetworkRule.md]: @lewu-msft please fill in the template sections provided (synopsis, description, examples)

File: src/ResourceManager/DataLakeStore/Commands.DataLakeStore/help/Get-AzureRmDataLakeStoreVirtualNetworkRule.md [hunk 9]
```diff
@@ -0,0 +1,108 @@
+---
+external help file: Microsoft.Azure.Commands.DataLakeStore.dll-Help.xml
+Module Name: AzureRM.DataLakeStore
+online version:
```
Comment (cormacpayne) [inline src/ResourceManager/DataLakeStore/Commands.DataLakeStore/help/Get-AzureRmDataLakeStoreVirtualNetworkRule.md]: @lewu-msft 

> online version: https://docs.microsoft.com/en-us/powershell/module/azurerm.datalakestore/get-azureatalakestorevirtualnetworkrule

File: src/ResourceManager/DataLakeStore/Commands.DataLakeStore/help/Get-AzureRmDataLakeStoreVirtualNetworkRule.md [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+schema: 2.0.0
+---
+
+# Get-AzureRmDataLakeStoreVirtualNetworkRule
+
+## SYNOPSIS
+{{Fill in the Synopsis}}
```
Comment (cormacpayne) [inline src/ResourceManager/DataLakeStore/Commands.DataLakeStore/help/Get-AzureRmDataLakeStoreVirtualNetworkRule.md]: @lewu-msft same comment about filling in the template sections in this file

File: src/ResourceManager/DataLakeStore/Commands.DataLakeStore/help/Remove-AzureRmDataLakeStoreVirtualNetworkRule.md [hunk 11]
```diff
@@ -0,0 +1,157 @@
+---
+external help file: Microsoft.Azure.Commands.DataLakeStore.dll-Help.xml
+Module Name: AzureRM.DataLakeStore
+online version:
```
Comment (cormacpayne) [inline src/ResourceManager/DataLakeStore/Commands.DataLakeStore/help/Remove-AzureRmDataLakeStoreVirtualNetworkRule.md]: @lewu-msft 

> online version: https://docs.microsoft.com/en-us/powershell/module/azurerm.datalakestore/remove-azureatalakestorevirtualnetworkrule

File: src/ResourceManager/DataLakeStore/Commands.DataLakeStore/help/Remove-AzureRmDataLakeStoreVirtualNetworkRule.md [hunk 12]
```diff
@@ -0,0 +1,157 @@
...
+schema: 2.0.0
+---
+
+# Remove-AzureRmDataLakeStoreVirtualNetworkRule
+
+## SYNOPSIS
+{{Fill in the Synopsis}}
```
Comment (cormacpayne) [inline src/ResourceManager/DataLakeStore/Commands.DataLakeStore/help/Remove-AzureRmDataLakeStoreVirtualNetworkRule.md]: @lewu-msft same comment about filling out the template sections in this file

File: src/ResourceManager/DataLakeStore/Commands.DataLakeStore/help/Set-AzureRmDataLakeStoreVirtualNetworkRule.md [hunk 13]
```diff
@@ -0,0 +1,155 @@
+---
+external help file: Microsoft.Azure.Commands.DataLakeStore.dll-Help.xml
+Module Name: AzureRM.DataLakeStore
+online version:
```
Comment (cormacpayne) [inline src/ResourceManager/DataLakeStore/Commands.DataLakeStore/help/Set-AzureRmDataLakeStoreVirtualNetworkRule.md]: @lewu-msft 

> online version: https://docs.microsoft.com/en-us/powershell/module/azurerm.datalakestore/set-azureatalakestorevirtualnetworkrule

File: src/ResourceManager/DataLakeStore/Commands.DataLakeStore/help/Set-AzureRmDataLakeStoreVirtualNetworkRule.md [hunk 14]
```diff
@@ -0,0 +1,155 @@
...
+schema: 2.0.0
+---
+
+# Set-AzureRmDataLakeStoreVirtualNetworkRule
+
+## SYNOPSIS
+{{Fill in the Synopsis}}
```
Comment (cormacpayne) [inline src/ResourceManager/DataLakeStore/Commands.DataLakeStore/help/Set-AzureRmDataLakeStoreVirtualNetworkRule.md]: @lewu-msft same comment about filling out the template sections in this file

File: src/ResourceManager/DataLakeStore/Commands.DataLakeStore/packages.config [hunk 15]
```diff
@@ -1,6 +1,9 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
   <package id="Microsoft.Azure.DataLake.Store" version="1.1.9" targetFramework="net452" />
-  <package id="Microsoft.Azure.Management.DataLake.Store" version="2.3.0-preview" targetFramework="net452" />
+  <package id="Microsoft.Azure.Management.DataLake.Store" version="2.4.1-preview" targetFramework="net452" />
+  <package id="Microsoft.Rest.ClientRuntime" version="2.3.12" targetFramework="net452" />
```
Comment (cormacpayne) [inline src/ResourceManager/DataLakeStore/Commands.DataLakeStore/packages.config]: @lewu-msft same comment about removing the extra references from this file

File: src/ResourceManager/Profile/Commands.Profile/Commands.Profile.csproj [hunk 16]
```diff
@@ -121,6 +121,11 @@
       <LastGenOutput>Resources.Designer.cs</LastGenOutput>
     </EmbeddedResource>
   </ItemGroup>
+  <ItemGroup>
```
Comment (cormacpayne) [inline src/ResourceManager/Profile/Commands.Profile/Commands.Profile.csproj]: @lewu-msft please revert the changes made in this file

File: src/ResourceManager/Profile/Commands.Profile/packages.config [hunk 17]
```diff
@@ -8,6 +8,7 @@
   <package id="Microsoft.Azure.Gallery" version="2.6.2-preview" targetFramework="net45" />
   <package id="Microsoft.Azure.KeyVault.Core" version="1.0.0" targetFramework="net452" />
   <package id="Microsoft.Azure.Management.Authorization" version="2.0.0" targetFramework="net45" />
+  <package id="Microsoft.Azure.Management.DataLake.Store" version="2.4.1-preview" targetFramework="net452" />
```
Comment (cormacpayne) [inline src/ResourceManager/Profile/Commands.Profile/packages.config]: @lewu-msft please revert the changes made in this file

**Comment by lewu-msft** [comment]
No need to review this update for now. but I'm getting CommandNotFoundException : The term 'Test-DataLakeStoreVirtualNetwork' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again. Will update PR after resolving this issue. Thanks

**Comment by cormacpayne** [comment]
@lewu-msft you're getting this error when trying to record the test? I've seen this happen when the execution policy for _either_ 32 or 64-bit version of PowerShell isn't set accordingly -- you can run `Set-ExecutionPolicy -ExecutionPolicy Unrestricted` to resolve this.

File: src/ResourceManager/DataLakeStore/Commands.DataLakeStore.Test/Commands.DataLakeStore.Test.csproj [hunk 18]
```diff
@@ -90,6 +88,9 @@
     <None Include="SessionRecords\Microsoft.Azure.Commands.DataLakeStore.Test.ScenarioTests.AdlsAliasTests\TestAdlsFirewallRules.json">
       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
     </None>
+    <None Include="SessionRecords\Microsoft.Azure.Commands.DataLakeStore.Test.ScenarioTests.AdlsAliasTests\TestAdlsVirtualNetworkRules.json">
```
Comment (cormacpayne) [inline src/ResourceManager/DataLakeStore/Commands.DataLakeStore.Test/Commands.DataLakeStore.Test.csproj]: @lewu-msft the build is currently failing because the `.json` files you added a reference to in this file weren't added in the PR

**Comment by lewu-msft** [comment]
> @lewu-msft you're getting this error when trying to record the test? I've seen this happen when the execution policy for _either_ 32 or 64-bit version of PowerShell isn't set accordingly -- you can run `Set-ExecutionPolicy -ExecutionPolicy Unrestricted` to resolve this.

@cormacpayne Yes, got this error when trying to record the test. All cmdlet tests(including those I didn't touch) failed with the same CommandNotFoundException. I guess maybe I shouldn't remove all those extra references (`Microsoft.Rest.ClientRuntime, Microsoft.Rest.ClientRuntime.Azure, System.Runtime`)

File: src/ResourceManager/DataLakeStore/Commands.DataLakeStore.Test/Commands.DataLakeStore.Test.Netcore.csproj [hunk 19]
```diff
@@ -24,7 +24,8 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Azure.Management.DataLake.Store" Version="2.3.0-preview" />
+    <PackageReference Include="Microsoft.Azure.Management.DataLake.Store" Version="2.4.1-preview" />
```
Comment (cormacpayne) [inline src/ResourceManager/DataLakeStore/Commands.DataLakeStore.Test/Commands.DataLakeStore.Test.Netcore.csproj]: @lewu-msft be sure to also bump this version in `Commands.DataLakeStore.Netcore.csproj` -- this is why the Travis CI job is currently failing

**Comment by cormacpayne** [comment]
@lewu-msft just double-checking, you updated the execution policy for _both_ 32-bit and 64-bit versions of PowerShell, correct?

**Comment by lewu-msft** [comment]
> @lewu-msft just double-checking, you updated the execution policy for _both_ 32-bit and 64-bit versions of PowerShell, correct?

Yes, both are Unrestricted

**Comment by lewu-msft** [comment]
@cormacpayne Looks like after pushing, it doesn't show test failure error. Just `c:\workspace\powershell\build.proj(278,5): error MSB3073: The command ""C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe" -NonInteractive -NoLogo -NoProfile -Command "$ProgressPreference = 'SilentlyContinue';. c:\workspace\powershell\tools\GenerateHelp.ps1 -ValidateMarkdownHelp -GenerateMamlHelp -BuildConfig Debug -FilteredModules 'AzureRM.DataLakeStore' "" exited with code 1.` Is it just my local machine config issue?

File: src/ResourceManager/DataLakeStore/Commands.DataLakeStore.Test/ScenarioTests/AdlsAliasTests.ps1 [hunk 20]
```diff
@@ -6,7 +6,7 @@ function Test-DataLakeStoreTrustedIdProvider
 {
     param
 	(
-		$location = "West US"
```
[the remark this answers is not in the document]
Comment (lewu-msft) [reply src/ResourceManager/DataLakeStore/Commands.DataLakeStore.Test/ScenarioTests/AdlsAliasTests.ps1]: @cormacpayne  Thanks! It is this issue. New PR is created [here](https://github.com/Azure/azure-powershell/pull/7404)
